inventory-repository/src/main/scala/com/normation/inventory/ldap/core/InventoryMapper.scala

```diff
@@ -775,10 +775,63 @@ class InventoryMapper(
     }
   }
 
-    def nodeFromEntry(entry:LDAPEntry) : Box[NodeInventory] = {
-    def missingAttr(attr:String) : String = "Missing required attribute %s".format(attr)
+  def mapOsDetailsFromEntry(entry: LDAPEntry) : Box[OsDetails] = {
     def requiredAttr(attr:String) = entry(attr) ?~! missingAttr(attr)
+    for {
+      osName        <- requiredAttr(A_OS_NAME)
+      osVersion     <- requiredAttr(A_OS_VERSION).map(x => new Version(x))
+      kernelVersion <- requiredAttr(A_OS_KERNEL_VERSION).map(x => new Version(x))
+      osFullName    =  entry(A_OS_FULL_NAME).getOrElse("")
+      osServicePack =  entry(A_OS_SERVICE_PACK)
+      osDetails     <- if(entry.isA(OC_WINDOWS_NODE)) {
+                          val os = osName match {
+                            case A_OS_WIN_XP => WindowsXP
+                            case A_OS_WIN_VISTA => WindowsVista
+                            case A_OS_WIN_SEVEN => WindowsSeven
+                            case A_OS_WIN_2000 => Windows2000
+                            case A_OS_WIN_2003 => Windows2003
+                            case A_OS_WIN_2008 => Windows2008
+                            case A_OS_WIN_2008_R2 => Windows2008R2
+                            case A_OS_WIN_2012 => Windows2012
+                            case _ => UnknownWindowsType
```

Comment:
shouldn't this be externalized somewhere, so that when we add a new OS, we don't need to remember where to change the mapping ?